libsel4vmmplatsupport/src/arch/arm/guest_image.c

```diff
@@ -134,7 +134,14 @@ static int get_guest_image_type(const char *image_name, enum img_type *image_typ
 
 static int guest_write_address(vm_t *vm, uintptr_t paddr, void *vaddr, size_t size, size_t offset, void *cookie)
 {
-    memcpy(vaddr, cookie + offset, size);
+    int *fd = (int *)cookie;
```

Comment:
```suggestion
    int fd = *((int *)cookie);
```
Let's unwrap this directly, then we can use `fd` everywhere and not `fd[0]`. 
Or actually, it might be the simplest thing f we pass the value of `fd` directly in the cookie?

Reply:
Updated to grab the contents of the cookie as `fd`.